src/app/services/soum-data.service.js

```diff
@@ -0,0 +1,108 @@
+/* ngInject */
+export function SoumData($log, $http, $q, Config) {
+  const username = 'adbmongolia';
+
+  return {
+    load: soumId => {
+      // Return a promise that resolves with the constructed soum
+      return $q((resolve, reject) => {
+        // Send a HTTP request to load the soum's raw data
+        loadSoumData(soumId).then(responses => {
+          // Pass all response objects to merge them together as one row
+          const combinedRow = mergeResponses(responses);
+          // Format the combined result with the same structure as Config
+          const soum = formatSoumData(combinedRow);
+          // Done!
+          resolve(soum);
+        }, () => {
+          reject();
+        });
+      });
+    }
+  };
+
+  function mergeResponses(responses) {
+    // Take several CartoSQL responses (Assuming a single row each) and return
+    //  an object with the combined columns from each. Repeated column names
+    //  are overwritten.
+    const result = {};
+    for (let i = 0; i < responses.length; i++) {
```

Comment:
Haven't used it before, but you could potentially replace these for loops with ES6 for...of syntax

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of